jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/TFRecordExecutor.scala

```diff
@@ -22,9 +24,35 @@ import org.tensorflow.{SavedModelBundle, Tensor}
  *                    └── variables.index
  * @param config instantiated struct with the values from the feature_config.yaml relevant for inference
  */
+
+
 class TFRecordExecutor(dirPath: String, config: ModelExecutorConfig) {
+println(s"in TFRecordExecutor $dirPath")
   private val savedModelBundle = SavedModelBundle.load(dirPath, "serve")
+  println("\nsavedModelBundle:\n")
+  savedModelBundle.graph().operations().asScala.foreach(op => println(op.name()))
+
+  //val metaGraphDef = MetaGraphDef.parseFrom(savedModelBundle.metaGraphDef())
+  //val signatureMap = metaGraphDef.getSignatureDefMap
+  //val signatureKey = "serving_tfrecord"
+  //val signatureDef = signatureMap.get(signatureKey)
+
+  //val inputInfo = signatureDef.getInputsMap.asScala
+  //val outputInfo = signatureDef.getOutputsMap.asScala
+
+
+
   private val session = savedModelBundle.session()
+
+  // Initialize variables if an init op exists
+    try {
+      session.runner().addTarget("init_op").run() // Replace "init_op" with actual name
+    } catch {
+      case e: Exception => println("No init_op found or initialization failed.")
+    }
+
+  // Run the initialization operation
```

Comment:
@mohazahran Do you want to clean this up or let it be for context?

Reply:
I want to leave it so that we can track debugging attempts. I will add comments.